kernel/umask.c

```diff
@@ -0,0 +1,21 @@
+#include <myst/syscall.h>
+#include <myst/spinlock.h>
+#include <myst/thread.h>
+#include <myst/eraise.h>
+
+long myst_syscall_umask(mode_t mask)
```

Comment:
Don't feel we should dedicate a file just for a single function. The cons are increased compile time and decreased code readability.

Maybe move this and a few other file permission related functions to a separate file?

Reply:
Fixed. I moved it to kernel/fs.c (since it is file-system related).